### ISSUE TITLE
Feature request: s3fs reading from locally hosted service (minio) or other non-Amazon S3 compatible service

### ISSUE THREAD

**Issue by mskyttner** [duckdb/duckdb#2166]
**What does happen?**

It looks like using "s3://" against locally hosted S3 services or remote non-Amazon S3 compatible services (for example against S3 storage backed by minio or Digital Ocean Spaces or Archive.org S3 storage) is not supported. At least it looks so at https://github.com/duckdb/duckdb/blob/v0.2.8/extension/httpfs/s3fs.cpp#L76-L77 which appear to connect only against Amazon S3 storage services (host and https protocol are hardcoded?).

**What should happen?**

With the existing mechanism available for S3 connection settings (for example as described in https://github.com/duckdb/duckdb/issues/2123) , ie 

```
SET s3_region='es-east-1';
SET s3_access_key_id='<AWS access key id>';
SET s3_secret_access_key='<AWS secret access key>';
SELECT * FROM parquet_scan('s3://<bucket>/<file>');
```
... but complemented also with a setting for s3_endpoint_url (including protocol), reading against locally hosted S3 services (using for example minio) or non-Amazon remote S3 services would be possible.

**To Reproduce**

This is more of a feature request than a bug.

The arrow libraries allows reading and writing parquet to non-Amazon S3 services, see for example https://ursalabs.org/arrow-r-nightly/articles/fs.html#file-systems-that-emulate-s3 (for reading) and see https://gist.github.com/rkaneko/dd2fae35149a29405d5e287ccd62677f#file-sample-py-L23-L27 which shows how pyarrow configures the S3 connection for writing data. 

**Environment (please complete the following information):**
 - OS: all supported variants
 - DuckDB Version 0.2.28 (using duckdb CLI)

**Comment by hannesmuehleisen** [comment]
Great idea, happy to review a PR on this!

**Comment by tekumara** [comment]
FYI This was added in https://github.com/duckdb/duckdb/pull/2394

**Comment by mskyttner** [comment]
Great! A simple test using an on-premise minio instance seems to work nicely for me, using:

```bash
v0.3.4 662041e2b
Enter ".help" for usage hints.
D .read init.sql
┌──────────────┐
│ count_star() │
├──────────────┤
│ 321076       │
└──────────────┘
Run Time: real 0.005 user 0.002454 sys 0.000000
install 'httpfs';
load 'httpfs';
set s3_endpoint='http://minio:9000';
set s3_access_key_id='minioadmin';
set s3_secret_access_key='minioadmin';
.timer on
select count(*) from 's3://demo/masterfile.parquet';
.shell more init.sql
```

The duckdb CLI v 0.3.4 with 'httpfs' extension installed and using `parquet_scan` against a parquet file from a remote minio instance worked fine for me, too. Due to not being on-premise, the same query against the same dataset then took 0.355 s ie 70x slower, due to more network latency, I guess.

**Comment by mskyttner** [comment]
I have not been able to [get globbing to work](https://github.com/duckdb/duckdb/discussions/3499#discussioncomment-2630768) yet.

**Comment by hannes** [comment]
Globbing has been merged in #3225
